agent/consul/server_test.go

```diff
@@ -2100,3 +2105,39 @@ func TestServer_hcpManager(t *testing.T) {
 	hcp1.AssertExpectations(t)
 
 }
+
+// goldenMarkdown reads and optionally writes the expected data to the goldenMarkdown file,
+// returning the contents as a string.
+func goldenMarkdown(t *testing.T, name, got string) string {
```

Comment:
Part of this I believe can call into `github.com/hashicorp/consul/internal/testing/golden`

Reply:
I started looking into this and fixing the `flag redefined: update` this causes is going to take quite a bit of work. For now, I'm going to leave this as is.